src/conchat.js

```diff
@@ -81,6 +93,37 @@ class Con {
 
     this.#sendMessage('messages', message);
   }
+
+  configUsername(username) {
+    if (this.#isStarted()) {
+      console.log('🚫 con.chat()을 실행해주세요.');
+
+      return;
+    }
+
+    if (this.#hasUsername) {
+      console.log(`💁🏻 ${this.#username}님, 이미 이름을 설정하셨네요!`);
+
+      return;
+    }
+
+    (async () => {
```

Comment:
즉시 실행함수로 바로 실행시켜서 처리하는 것도 좋은 방법이지만
```suggestion
const checkUsername= async () => {
  const usersQuery = query(
    collection(store, 'users'),
    where('username', '==', username),
  );
  const userQuerySnapshot = await getDocs(usersQuery);
  const isUsernameExists = !userQuerySnapshot.empty;

  if (isUsernameExists) {
    console.log('🚫 이미 존재하는 이름입니다. 다시 설정해 주세요.');
  } else {
    this.#addUserToStore(username);

    console.log(`💁🏻 ${username}님 안녕하세요!`);
  }
}

checkUsername();
```
확장성을 고려했을 때, 함수로 만든 후 실행하는 것도 좋아보입니다.

그리고 즉시 실행함수는 1회성으로 한번 실행된다는 의도가 있습니다.
 코드의 모듈화, 내부 변수의 보호 반복문 내부 로직 비동기 처리를 위해서 사용하기에 함수화를 통해 분리하는 것도 고려해보면 좋을거 같습니다.
[즉시 실행 함수 사용시기 ](https://vinayak-hegde.medium.com/js-when-to-use-an-iife-603e83d81327)

Reply:
리뷰 감사합니다! 

제가 configUsername에서 즉시 실행 함수를 사용하여 
비동기 로직을 사용할 수 밖에 없었던 이유는, 
Promise 객체를 반환하지 않고 로직 내부에서 Promise 결과값을 활용하여 중복 검증을 하기 위함이었습니다.
함수를 만든 후 실행시키면 Promise 객체를 반환 후 해당 값의 Boolean 여부만 체크할 수 있지만 콘솔에 Promise 객체가 보이는 문제가 있었습니다. 

따라서 현 로직에서는 1회성으로 실행되는 경우가 맞으며 
은닉화에 대한 대안은 조금 더 고민 후 적절한 방안을 발견한다면 리팩터링 테스트에서 수정하도록 하겠습니다.
감사합니다!